docs/BuildFromSource.md

```diff
@@ -1,155 +1,201 @@
+## Step 3: Build the repo
 
-The [ASP.NET Core Module](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module) (ANCM) for IIS is not supported when running projects in this repository.
+Before opening our .sln/.slnf files in Visual Studio or VS Code, you will need to build the repo locally.
 
-After using `startvs.cmd` to open a solution in Visual Studio, the Kestrel web host option must be used (name of the project) and not IIS Express.
+### In Visual Studio
 
-Example of running the `MvcSandbox` project:
+To set up your project for development on Visual Studio, you'll need to execute the following command.
 
-`.\startvs.cmd .\src\Mvc\Mvc.sln`
+```ps1
+PS1> .\restore.cmd
+```
 
-![Web host options in Visual Studio](./vs-iis-express-aspnet-core-mvc-sandbox.jpg)
+> :bulb: If you happen to be working on macOS or Linux, you can use the `restore.sh` command.
+
+This will download the required tools and restore all projects inside the repository. At that point, you should be able
+to open the .sln file or one of the project specific .slnf files to work on the projects you care about.
+
+   > :bulb: Pro tip: you will also want to run this command after pulling large sets of changes. On the master
+   > branch, we regularly update the versions of .NET Core SDK required to build the repo.
+   > You will need to restart Visual Studio every time we update the .NET Core SDK.
+
+To open the solution file in Visual Studio, run the following command.
+
+```ps1
+PS1> .\startvs.cmd
+```
+
+This will open the solution file for the entire repo. Typically, you want to focus on a single project within the monorepo. For example,
+if you want to work on Blazor WebAssembly, you'll need to launch the solution file for that project by changing into the `src/Components`
+directory and executing `startvs.cmd` in that directory like so:
 
-## Building with Visual Studio Code
+```ps1
+PS1> cd src\Components
+PS1> .\startvs.cmd
+```
+
+After opening the solution in Visual Studio, you can build/rebuild using the controls in Visual Studio.
+
+#### A brief interlude on solution files
+
+We have a single .sln file for all of ASP.NET Core, but most people don't work with it directly because Visual Studio
+doesn't currently handle projects of this scale very well.
+
+Instead, we have many Solution Filter (.slnf) files which include a sub-set of projects. See the Visual Studio
+documentation [here](https://docs.microsoft.com/en-us/visualstudio/ide/filtered-solutions?view=vs-2019) for more
+information about Solution Filters.
+
+These principles guide how we create and manage .slnf files:
+
+1. Solution files are not used by CI or command line build scripts. They are meant for use by developers only.
+2. Solution files group together projects which are frequently edited at the same time.
+3. Can't find a solution that has the projects you care about? Feel free to make a PR to add a new .slnf file.
```

Comment:
We should probably make a note that VS 4 Mac doesn't currently support slnf files, but that work is on their backlog, and will hopefully be coming soon.

Reply:
Added.